### DISCUSSION TITLE
How to Fix Error "Closed Closed" or other install issues

### DISCUSSION THREAD

**Discussion by BlackwolfSGP** [jhassan8/crunchyroll-tizen#70] [Show and tell]
I can't guarantee this is THE solution for everyone, but seems like this is how it works. Essentially, when you get the error "closed closed" your TV is a lower version SDK than the current one, as of the time of this writing the latest is V7. To fix this you have to download the correct sdk version that corresponds to your TV.

[This post/page](https://blog.infernored.com/how-to-get-back-tizen-studio-1-3-and-tv-extensions-3-0/) Explains how to add older SDKs to your Tizen Studio, it was made for older versions but still works in 2023.
[This page](https://developer.samsung.com/smarttv/develop/specifications/tv-model-groups.html) can help you determine which version you need, although device manager will show it as well.
and finally, [This page](https://developer.samsung.com/smarttv/develop/tools/tv-extension/archive.html) has the repository from samsung for all the versions available, which are all V3 onward, v1 and v2 samsung has discontinued for security reasons. 

I hope this helps everyone who has been having issues getting it to run. Also remember you need to setup your certificate, and THEN permit install application in device manager. By this point the error is no longer closed closed, instead you get a fail error that says "it cannot be installed at this time, try again later". 

If anyone else has found other causes I would love to collect them in this discussion as well

**Comment by RK721** [comment]
Crazy how I spent all morning solving this when I could've just came here lol.

**Reply by BlackwolfSGP** [reply]
Sad thing about samsung's choice to use their weird environment instead of something easy.

**Reply by heymrlen** [reply]
@RK721 where you able to install it on Tizen 2.4? If yes, could you please share the steps that you've followed? Thanks!
